### DISCUSSION TITLE
Theme doesn't show up in the GUI of the docker image

### DISCUSSION THREAD

**Discussion by henkiewie** [Lissy93/dashy#563] [Help and Support] [answered]
Hi,

I'm on the latest image (2.0.5).

I tought i  wanted to save my own theme somewhere so it can not get overwritten by a new docker image or by mistake. And make some changes to the layout through the interface because I am not a css guy 😄.  
I copied the adventure theme (because it hides a lot of the background properties) from the github resource to a file and copied it to a scss file on my host and mounted it as explianed in the doc. That worked OK because I can see the content in a docker exec console session in the file `/app/src/styles/user-defined-themes.scss`
I also gave it a new name in the header `html[data-theme="Newone"] {` and added that name in `appConfig.cssThemes` saved the config. It did a rebuild (no errors).
But then I should see the new name in the drop-down box for the themes in th GUI, but that does not seem to happen.  It also does not show up in the log that it is loaded somewhere (should it?)
I also changed the owner to root instead of the default user pi. But the result is still the same.

Where did I go wrong ❓ 

regards,
Henk

**Comment by henkiewie** [comment]
When I open an Incognito browser it appears in the theme name field.

![image](https://user-images.githubusercontent.com/74558874/158331084-2810ba6a-2a09-4664-8d5e-6230303007de.png

But when i change it to another theme it is not coming back. It even changes the name in the conf.yml to the new one.

![image](https://user-images.githubusercontent.com/74558874/158331394-8f661a6a-4766-47fd-87c2-7c8e594ba2ec.png)
😕

**Comment by lissy93** [comment]
I think the same issue as #560
But I will look into it...

**Reply by emiran-orange** [reply]
Hi, 
I use 2.0.7 in a container, user-defined theme is well loaded when setting the new theme in `appConfig.theme` (and `appConfig.cssThemes`) but it does not appear in dropdown list (and once I change theme, I cannot change back).
Not a vue dev but it may have smthg to do with `extraThemeNames` not being part of `themeNames` in src/components/Settings/ThemeSelector.vue.
If you confirm, I can drop a PR

**Reply by lissy93** [reply]
Oh, you're correct, I'd not noticed that, good spot!
Replacing with [this line](https://github.com/lissy93/dashy/blob/master/src/components/Settings/ThemeSelector.vue#L78) with:
```javascript
return [
  ...this.extraThemeNames, ...externalThemeNames, ...Defaults.builtInThemes, ...specialThemes,
];
``` 

Once updated, that will add any themes in `appConfig.cssThemes` to the theme dropdown. And once a theme is applied, it adds `data-theme="my-theme"` to the html tag. Which can then be targeted with `html[data-theme='my-theme'] { ... }`.

This should work if custom styles are in [`user-defined-themes.scss`](https://github.com/Lissy93/dashy/blob/master/src/styles/user-defined-themes.scss), but the external stylesheet option is broken which I think was the original issue, so I need to change how it works.

**Reply by emiran-orange** [reply]
Do you want the PR in the meantime ?

**Reply by lissy93** [reply]
That would be awesome if you could :)
And I will work on the CSS imports, once I've merged #606 and #617

**Reply by emiran-orange** [reply]
That's it: https://github.com/Lissy93/dashy/pull/619

**Comment by henkiewie** [comment]
🤣  The suspense is killing me. No serious I think this will help a lot of docker users who want to start making their own templates (like me). 👍

**Comment by henkiewie** [comment, marked as answer]
Problem is fixed. Thank you @emiran-orange and @Lissy93 
![Screenshot 2022-04-30 at 08 19 29](https://user-images.githubusercontent.com/74558874/166094386-782475d6-71c5-4bf0-bdb2-2fb7f391e4bc.png)

**Reply by lissy93** [reply]
I actually wasn't expecting that alone to fix it 😆 But I'm glad it's working
P.s awesome theme :) When you're finished configuring, would you be up for sharing it on the [Showcase](https://github.com/Lissy93/dashy/blob/master/docs/showcase.md)?

**Reply by henkiewie** [reply]
I did just now but your bot keeps closing my submits.

**Reply by lissy93** [reply]
Sorry about that, added it in 6018a49bed9415d29b392e162a69db03b21f8d5d :) 
The Grafana kiosk mode widgets are a great idea, they look awesome too!
